### ISSUE TITLE
Support for local DuckDB files

### ISSUE THREAD

**Issue by wearpants** [hasura/ndc-duckdb#2]
It would be nice to be able to open a local duckdb file, not just motherduck. I'm looking to use Hasura in a manufacturing monitoring/analytics project with a client, and the network is air gapped, i.e. no Internet access.

I generate new read only duckdb files every few minutes to the same path - it would be nice if Hasura could detect the new file (via `stat` or timestamp) & transparently reopen the file to pick up new data.

**Comment by teaguesterling** [comment]
I was looking for the same thing as you. I took a stab at getting something working based on the SQLite DataConnector Agent (which may have been a poor choice to deviate from this repo). It's FAR from ready but I was able to get it up and running with only a few minor changes (and some major packaging issues). I'd be happy to hear if it works for you https://github.com/teaguesterling/duckdb-dataconnector-agent/. Of course, the most useful feature (of exposing views that point at local files) isn't working yet. But there's a path forward that doesn't seem too out-of- reach.
